include/envoy/event/dispatcher.h

```diff
@@ -32,6 +32,16 @@ class Dispatcher {
 public:
   virtual ~Dispatcher() {}
 
+  /**
+   * Returns a time-source to use with this dispatcher.
+   *
+   * TODO(#4160) the implementations currently manage timer events that
+   * ignore the time-source, and thus can't be mocked or faked. So it's
+   * difficult to mock time in an integration test without mocking out
+   * the dispatcher.
```

Comment:
Is the only benefit here then of providing the `timeSource()` to allow for convenient threading of this object across any object that needs time and already has a dispatcher reference?